### ISSUE TITLE
cant make mode directed tabling example work (Levenshtein distance)

### ISSUE THREAD

**Issue by jburse** [SWI-Prolog/swipl-devel#490]
Was trying to run this example, as mode directed tabling:

![lev_min3](https://user-images.githubusercontent.com/5411487/62012606-af866580-b188-11e9-93ed-894006468ac4.PNG)
https://en.wikipedia.org/wiki/Levenshtein_distance#Definition

But somehow it fails with no answer. I only get this result:

![image](https://user-images.githubusercontent.com/5411487/62012522-b95b9900-b187-11e9-8833-d5ea233bfd3b.png)

The program code and query are:

```
:- table lev(_,_,po(>)).
lev([], [], 0).
lev([X|L], [X|R], D) :- lev(L, R, D).
lev([_|L], R, D) :- lev(L, R, H), D is H+1.
lev(L, [_|R], D) :- lev(L, R, H), D is H+1.

?- set_prolog_flag(double_quotes, codes).
true.

?- lev("godelxbach","gyodelbach",M).
false.
```

**Comment by jburse** [comment]
I had more luck in YAP Prolog, although YAP Prolog has a 
slightly different behaviour, it also shows approximate solutions:

![image](https://user-images.githubusercontent.com/5411487/62012541-e3ad5680-b187-11e9-892c-be13c0d5ec9d.png)

And it also worked in my own prototype system:

![image](https://user-images.githubusercontent.com/5411487/62012554-0c355080-b188-11e9-91a3-f51166a3eb87.png)

**Comment by jburse** [comment]
Maybe a problem of wrong aggregate functions. Is min supported in mode directed tabling? I made a different test, and it worked:

:- table path(_,_,lattice(combine)).
path(X, X, []).
path(X, Y, [X|L]) :-
   edge(X, Z), path(Z, Y, L).

% edge(-Vertex, -Vertex)
edge(a, b).         edge(a, c).
edge(a, d).         edge(b, c).
edge(c, d).         edge(c, e).
edge(d, e).

% combine(+List, +List, -List)
combine(X, Y, Z) :- length(X, N), length(Y, M),
   (N < M -> Z = X; Z = Y).

**Comment by JanWielemaker** [comment]
Fixed with 511594ae56c7ef35de1263f698ea4b73a6419f73.  Using either `min` or `po(>)`, the answer is now `2`.
